x-pack/plugins/osquery/public/types.ts

```diff
@@ -20,8 +20,10 @@ import { getLazyOsqueryAction } from './shared_components';
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface OsqueryPluginSetup {}
+
 export interface OsqueryPluginStart {
   OsqueryAction?: ReturnType<typeof getLazyOsqueryAction>;
+  osqueryMenuItem: (props: { agentId: string; onClick: () => void }) => React.ReactElement | null;
```

Comment:
I don't think our plugin should be aware of `onClick` callbacks

Reply:
removed 👍